File: javaapp/src/main/java/com/javapp/datetime/gigasecond/Gigasecond.java
package com.javapp.datetime.gigasecond;

import java.math.BigInteger;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    //A java.javapp.gigasecond is one billion (10**9) seconds.
    public static final Long GIGASECOND = BigInteger.valueOf(10).pow(9).longValue();
    private LocalDateTime localDate;

    /*constructor*/
    public Gigasecond(LocalDate localDate) {
        this(localDate.atStartOfDay());
    }

    /*constructor*/
    public Gigasecond(LocalDateTime localDate) {
        this.localDate = localDate;
    }

    public LocalDateTime getDate() {
        return localDate.plusSeconds(GIGASECOND);
    }
/*CLASS END*/
}

File: kttime/src/main/kotlin/com/kotlinground/kttime/gigasecond/Gigasecond.kt
package com.kotlinground.kttime.gigasecond

import java.time.LocalDate
import java.time.LocalDateTime

/**
 * @author lusinabrian on 20/05/17.
 */
data class Gigasecond(val initialDateTime: LocalDateTime) {
    constructor(initialDateTime: LocalDate) : this(initialDateTime.atTime(0, 0))

    val date = initialDateTime.plusSeconds(1000000000)
}